### PR TITLE
Add JAVA_OPTS and JAVA_TOOL_OPTIONS to envoverwriter

### LINKED ISSUE #1130: JAVA_OPTS is overriden during instrumentation
**Describe the bug**
Somewhat similar to https://github.com/keyval-dev/odigos/issues/1083 but for JAVA

**To Reproduce**

Prior to the instrumentation, we had in the entrypoint.sh
```
  JAVA_OPTS="$JAVA_OPTS
            --add-opens java.base/java.time=ALL-UNNAMED
```
which is needed for packages like java.time based on https://openjdk.org/jeps/396 with java17

but after the instrumentation
```
$ set|grep JAVA
JAVA_OPTS='-javaagent:/var/odigos/java/javaagent.jar -Dotel.traces.sampler=always_on -Dotel.logs.exporter=none -Dotel.exporter.otlp.protocol=grpc -Dotel.exporter.otlp.endpoint=http://192.168.1.112:4317/'
JAVA_TOOL_OPTIONS='-javaagent:/var/odigos/java/javaagent.jar -Dotel.traces.sampler=always_on -Dotel.logs.exporter=none -Dotel.exporter.otlp.protocol=grpc -Dotel.exporter.otlp.endpoint=http://192.168.1.112:4317/'
```
which results in
```
2024-04-18 13:33:58,966  WARN [qtp1662844977-6748] [DefaultHandlerExceptionResolver]   : Resolved [org.springframework.http.converter.HttpMessageNotWritableException: Could not write JSON: Unable to make field private final java.time.LocalDate java.time.LocalDateTime.date accessible: module java.base does not "opens java.time" to unnamed module @d453a986; nested exception is java.lang.reflect.InaccessibleObjectException: Unable to make field private final java.time.LocalDate java.time.LocalDateTime.date accessible: module java.base does not "opens java.time" to unnamed module @d453a986]
```

**Expected behavior**
It would be preferable to either append the JAVA_OPTS or have the options of not settings it

**Additional context**
JAVA_OPTS maybe set in the dockerfile again and not in the deployment/container env variables, unfortunate, but again, we may need to find a way to discover it from the process, similar to PYPATH

### PR DESCRIPTION
**Pull request by RonFed**
Resolves #1130

### CODE REVIEW

**Comment by RonFed** [comment]
- [ ] Remove `ShouldRevert` since we are using the annotation for original env var value.
- [ ]  Allow to define overwrite value per Sdk (internal map per env var)

**Review by edeNFed — APPROVED** [review]

File: common/envOverwrite/overwriter.go [hunk 1]
```diff
@@ -1,71 +1,150 @@
...
 	}
 
-	if !strings.Contains(value, valToAppend.Value) {
+	val, ok := env.values[sdk]
+	if !ok {
+		// We don't care about this environment variable for this SDK
+		return false
+	}
+
+	if !strings.Contains(value, val) {
```
Comment (blumamir) [inline common/envOverwrite/overwriter.go:98]: check all SDKs

**Review by blumamir — APPROVED** [review]

File: common/envOverwrite/overwriter.go [hunk 2]
```diff
@@ -1,71 +1,150 @@
...
+}
+// EnvValuesMap is a map of environment variables odigos uses for various languages and goals.
 // The key is the environment variable name and the value is the value to be set or appended
 // to the environment variable. We need to make sure that in case any of these environment
 // variables is already set, we append the value to it instead of overwriting it.
-var EnvValues = map[string]struct {
-	Value string
-	Delim string
-}{
+var EnvValuesMap = map[string]envValues{
```
Comment (blumamir) [inline common/envOverwrite/overwriter.go:17]: perhaps add a note that these values should be correlated to the ones in the dockerfile

File: common/otelsdk.go [hunk 3]
```diff
@@ -24,3 +24,14 @@ type OtelSdk struct {
...
 }
+
+var (
+	OtelSdkNativeCommunity  = OtelSdk{SdkType: NativeOtelSdkType, SdkTier: CommunityOtelSdkTier}
+	OtelSdkEbpfCommunity    = OtelSdk{SdkType: EbpfOtelSdkType, SdkTier: CommunityOtelSdkTier}
+	OtelSdkNativeEnterprise = OtelSdk{SdkType: NativeOtelSdkType, SdkTier: EnterpriseOtelSdkTier}
+	OtelSdkEbpfEnterprise   = OtelSdk{SdkType: EbpfOtelSdkType, SdkTier: EnterpriseOtelSdkTier}
+)
+
+func NewOtelSdk(sdkType OtelSdkType, sdkTier OtelSdkTier) OtelSdk {
```
[the remark this answers is not in the document]
Comment (RonFed) [reply common/otelsdk.go:35]: removed

**Pull request merged**